examples/groupchat.c

```diff
@@ -0,0 +1,409 @@
+			return;
+		}
+
+		fprintf(stderr, "Invitation for %s: %s\n", arg, invitation);
+		free(invitation);
+	} else if(!strcasecmp(buf, "canonical")) {
+		bool set;
+		char *host = NULL, *port = NULL;
+
+		if(!arg) {
+			fprintf(stderr, "/canonical requires an argument!\n");
+			return;
+		}
+
+		if ( (0 == strncasecmp(arg, "-h", 2)) && (strlen(arg) > 2) ) {
+			host = arg + 2;
+		} else if ( (0 == strncasecmp(arg, "-p", 2)) && (strlen(arg) > 2) ) {
+			port = arg + 2;
+		} else {
+			fprintf(stderr, "Unknown argument: %s!\n", arg);
+			return;
+		}
+
+		if ( arg1 ) {
+			if ( (0 == strncasecmp(arg1, "-h", 2)) && (strlen(arg1) > 2) ) {
+				host = arg1 + 2;
+			} else if ( (0 == strncasecmp(arg1, "-p", 2)) && (strlen(arg1) > 2) ) {
+				port = arg1 + 2;
+			} else {
+				fprintf(stderr, "Unknown argument: %s!\n", arg1);
+				return;
+			}
+		}
+
+		if ( !host && !port ) {
+			fprintf(stderr, "Unable to set Canonical address because no valid arguments are found!\n");
+			return;
+		}
+
+		set = meshlink_set_canonical_address(mesh, meshlink_get_self(mesh), host, port);
+
+		if(!set) {
+			fprintf(stderr, "Could not set canonical address '%s:%s': %s\n", host, port, meshlink_strerror(meshlink_errno));
+			return;
+		}
+
+		fprintf(stderr, "Canonical address set as '%s:%s'\n", host, port);
+	} else if(!strcasecmp(buf, "group")) {
+		if(!arg) {
+			fprintf(stderr, "/group requires an argument!\n");
+			return;
+		}
+
+		if(!( s = meshlink_submesh_open(mesh, arg) ) ) {
+			fprintf(stderr, "Could not create group: %s\n", meshlink_strerror(meshlink_errno));
+		} else {
+			fprintf(stderr, "Group '%s' created!\n", s->name);
+		}
+	} else if(!strcasecmp(buf, "join")) {
+		if(!arg) {
+			fprintf(stderr, "/join requires an argument!\n");
+			return;
+		}
+
+		meshlink_stop(mesh);
+
+		if(!meshlink_join(mesh, arg)) {
+			fprintf(stderr, "Could not join using invitation: %s\n", meshlink_strerror(meshlink_errno));
+		} else {
+			fprintf(stderr, "Invitation accepted!\n");
+		}
+
+		if(!meshlink_start(mesh)) {
+			fprintf(stderr, "Could not restart MeshLink: %s\n", meshlink_strerror(meshlink_errno));
+			exit(1);
+		}
+	} else if(!strcasecmp(buf, "kick")) {
+		if(!arg) {
+			fprintf(stderr, "/kick requires an argument!\n");
+			return;
+		}
+
+		meshlink_node_t *node = meshlink_get_node(mesh, arg);
+
+		if(!node) {
+			fprintf(stderr, "Error looking up '%s': %s\n", arg, meshlink_strerror(meshlink_errno));
+			return;
+		}
+
+		meshlink_blacklist(mesh, node);
+
+		fprintf(stderr, "Node '%s' blacklisted.\n", arg);
+	} else if(!strcasecmp(buf, "who")) {
+		if(!arg) {
+			nodes = meshlink_get_all_nodes(mesh, nodes, &nnodes);
+
+			if(!nnodes) {
+				fprintf(stderr, "Could not get list of nodes: %s\n", meshlink_strerror(meshlink_errno));
+			} else {
+				fprintf(stderr, "%zu known nodes:", nnodes);
+
+				for(size_t i = 0; i < nnodes; i++) {
+					fprintf(stderr, " %s", nodes[i]->name);
```

Comment:
It would be nice to also print which submesh each node belongs to.

Reply:
In the latest commit iam also printing the submesh to which that node belongs to. For this i had to add the meshlink_get_all_nodes_by_submesh and meshlink_get_node_submesh api's in the latest commit